src/routes/index.js

```diff
@@ -187,12 +187,13 @@ if (config.IS_INTERNAL) {
       },
     }
 
-    console.log('create data session with:', data)
-
+    console.log('create data session with:', data) 
+    const apiKey = req.body.apiKey || config.IVY_API_KEY
+   
     const { data: session } = await axios
       .post(`${config.IVY_API_URL}/service/data/session/create`, data, {
         headers: {
-          'X-Ivy-Api-Key': config.IVY_API_KEY,
+          'X-Ivy-Api-Key': apiKey
```

Comment:
Shouldn't we have `config.IVY_API_KEY` as fallback

```suggestion
          'X-Ivy-Api-Key': apiKey ?? config.IVY_API_KEY
```